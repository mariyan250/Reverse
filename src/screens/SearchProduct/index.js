// React
import React, { useEffect } from 'react';

// Connect Redux
import { connect } from 'react-redux';
import { getPosts } from 'store/actions/post';

// Components
import Header from 'shared/components/Header';
import ItemList from './components/ItemList';

function SearchProduct({ getPosts, posts, location: { search } }) {
  useEffect(() => {
    getPosts(search);
  }, [getPosts, search]);

  return (
    <main className="search">
      <Header />
      <ItemList items={posts} />
    </main>
  );
}

const mapStateToProps = (state) => ({ posts: state.posts.items });

export default connect(mapStateToProps, { getPosts })(SearchProduct);
