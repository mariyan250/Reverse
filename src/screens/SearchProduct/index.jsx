// React
import React, { Fragment, useEffect, useState } from 'react';
import './index.scss';

// Redux
import { connect } from 'react-redux';
import { getPosts } from 'store/actions/post';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import Header from 'shared/components/Header';
import ItemList from 'shared/components/Product/CardList';
import Search from 'shared/components/Search';
import FilterButton from './components/FilterButton';

function SearchProduct({ getPosts, posts, location: { search } }) {
  const [open, setOpen] = useState(false);
  const { t } = useTranslation();

  useEffect(() => {
    getPosts(search);
  }, [getPosts, search]);

  return (
    <Fragment>
      <Header />

      <main className="search-product px-4 mx-auto">
        <section className="my-5 d-flex">
          <Search />
        </section>

        <section className="products-upper-line d-flex justify-content-between align-items-center mb-4">
          <h3>{t('search_screen.heading')}</h3>
          <FilterButton isOpen={open} onClick={() => setOpen(!open)} />
        </section>

        <section className="products-container app-bg-gray-darker p-2 p-md-4">
          <ItemList items={posts} title="search_screen.heading" />
        </section>
      </main>
    </Fragment>
  );
}

const mapStateToProps = (state) => ({ posts: state.posts.items });

export default connect(mapStateToProps, { getPosts })(SearchProduct);
