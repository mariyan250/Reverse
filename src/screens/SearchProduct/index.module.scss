@import '../../index.scss';

.search {
  max-width: $layout-max-width;

  .loading {
    margin-top: to-rem(150px);
  }
}
