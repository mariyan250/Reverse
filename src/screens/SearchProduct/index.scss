@import '../../index.scss';

.search-product {
  max-width: $layout-max-width;
}
