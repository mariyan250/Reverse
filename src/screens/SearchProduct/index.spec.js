import React from 'react';
import { render, waitForElement, cleanup } from 'tests/test-utils';
import SearchProduct from './index';

describe('SearchProduct', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  afterEach(cleanup);

  describe('Rendering', () => {
    it('should render the component with initial state', async () => {
      const { container, getByTestId } = render(
        <SearchProduct location={{ search: 'q=technology' }} />,
        initialState,
      );
      await waitForElement(() => getByTestId('search-product'));
      expect(container).toMatchSnapshot();
    });
  });
});
