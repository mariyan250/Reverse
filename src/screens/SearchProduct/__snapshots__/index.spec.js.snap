// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProduct should render the component 1`] = `
<div>
  <main
    class="search px-4 mx-auto"
  >
    <section
      class="my-5 d-flex"
    >
      <form
        action="/search"
        class="form w-100 mx-auto"
      >
        <article
          class="d-flex"
        >
          <input
            data-testid="searchInput"
            name="query"
            placeholder="home_screen.input.placeholder"
            required=""
            type="text"
            value=""
          />
          <article>
            <button
              class="h-100 border-0 app-bg-secondary d-flex justify-content-center align-items-center"
            >
              <i
                class="fas fa-search app-text-white"
              />
            </button>
          </article>
        </article>
      </form>
    </section>
    <section
      class="d-flex justify-content-between align-items-center mb-4"
    >
      <h3>
        search_screen.heading
      </h3>
      <section
        class="filter-button"
        role="filter-button"
      >
        <button
          class="box-shadow rounded d-flex p-2 justify-content-center align-items-center position-relative border-0 app-bg-white"
        >
          <p>
            search_screen.filter_button
          </p>
          <i
            class="fas fa-angle-down ml-3 false"
          />
        </button>
      </section>
    </section>
    <section
      class="app-bg-gray-darker p-2 p-md-4"
    >
      <ul
        class="list"
      >
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/1"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/31489/31488238/images/res_cf71f94f6a0749441e0c420bf54540c4.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Lenovo IdeaPad 3, 15.6, AMD Ryzen™
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Sofia
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                979.99
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/2"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/15874/15873942/images/res_b8032c207f90cec1988f271d669be260.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Lenovo G50-30, 15.6, Intel Pentium™
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Sofia
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                600
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/3"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/17043/17042927/images/res_b0edd1ced6577580eaec29aad246405d.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Iphone X, 128SD / 256SD, Black and White
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Blagoevgrad
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                1200
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;

exports[`SearchProduct should set filterButton to open when click 1`] = `
<div>
  <main
    class="search px-4 mx-auto"
  >
    <section
      class="my-5 d-flex"
    >
      <form
        action="/search"
        class="form w-100 mx-auto"
      >
        <article
          class="d-flex"
        >
          <input
            data-testid="searchInput"
            name="query"
            placeholder="home_screen.input.placeholder"
            required=""
            type="text"
            value=""
          />
          <article>
            <button
              class="h-100 border-0 app-bg-secondary d-flex justify-content-center align-items-center"
            >
              <i
                class="fas fa-search app-text-white"
              />
            </button>
          </article>
        </article>
      </form>
    </section>
    <section
      class="d-flex justify-content-between align-items-center mb-4"
    >
      <h3>
        search_screen.heading
      </h3>
      <section
        class="filter-button"
        role="filter-button"
      >
        <button
          class="box-shadow rounded d-flex p-2 justify-content-center align-items-center position-relative border-0 app-bg-white"
        >
          <p>
            search_screen.filter_button
          </p>
          <i
            class="fas fa-angle-down ml-3 toggle"
          />
          <ul
            class="filter-dropdown position-absolute app-bg-white rounded-lg box-shadow overflow-hidden d-flex flex-column justify-content-center"
            data-testid="filter-dropdown"
          >
            <li
              class="py-3 text-center"
              role="filter-newest"
            >
              search_screen.filter_dropdown.newest
            </li>
            <li
              class="py-3 text-center"
              role="filter-lowest"
            >
              search_screen.filter_dropdown.lowest
            </li>
            <li
              class="py-3 text-center"
              role="filter-highest"
            >
              search_screen.filter_dropdown.highest
            </li>
          </ul>
        </button>
      </section>
    </section>
    <section
      class="app-bg-gray-darker p-2 p-md-4"
    >
      <ul
        class="list"
      >
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/1"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/31489/31488238/images/res_cf71f94f6a0749441e0c420bf54540c4.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Lenovo IdeaPad 3, 15.6, AMD Ryzen™
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Sofia
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                979.99
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/2"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/15874/15873942/images/res_b8032c207f90cec1988f271d669be260.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Lenovo G50-30, 15.6, Intel Pentium™
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Sofia
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                600
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
        <li
          class="item app-bg-white box-shadow overflow-hidden"
        >
          <a
            class="app-text-secondary"
            href="/item/3"
          >
            <section
              class="container d-flex overflow-hidden position-relative justify-content-center"
            >
              <img
                alt="search_screen.item.img.alt"
                class="img img-fluid position-absolute p-5"
                src="https://s13emagst.akamaized.net/products/17043/17042927/images/res_b0edd1ced6577580eaec29aad246405d.jpg"
              />
            </section>
            <section
              class="py-3 px-4 h-100 app-bg-gray"
            >
              <p
                class="name text-center mb-4"
              >
                Iphone X, 128SD / 256SD, Black and White
              </p>
              <p
                class="mb-3"
              >
                <i
                  class="fas fa-map-marker-alt mr-2"
                />
                Blagoevgrad
              </p>
              <p
                class="mb-4"
              >
                <i
                  class="fas fa-clock mr-2"
                />
                8 July - 6:00
              </p>
              <p
                class="price"
              >
                1200
                 
                general.currency
              </p>
            </section>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
